dali/python/nvidia/dali/ops.py

```diff
@@ -419,6 +419,14 @@ def __init__(self, **kwargs):
                 kwargs["dtype"] = kwargs["output_dtype"]
                 del kwargs["output_dtype"]
 
+            if "image_type" in kwargs.keys() and name in ("Crop", "Slice", "CropMirrorNormalize"):
+                with warnings.catch_warnings():
+                    warnings.simplefilter("default")
+                    warnings.warn(
+                        ("Argument name 'image_type' for operator {} is deprecated. " +
+                         "It should be removed from the argument list, as it is not necessary.").format(type(self).__name__),
+                        DeprecationWarning, stacklevel=2)
+                del kwargs["image_type"]
```

Comment:
I don't like this kind of specific warning being placed in generic place. I think that a mechanism for deprecation at schema level would be useful.

Reply:
Added a general solution